### PR TITLE
Add Piezo probe Stealthburner dragon toolhead

### PR DESCRIPTION
**Pull request by 4k0b**
* [ x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x ] Folder and file naming:
    * Folders and filenames shouldn't contain spaces. Only letters `a-zA-Z`, numbers `0-9`, underscores `_`, hyphens `-` and periods `.`
    * Primary color: `part_xyz.stl`
    * Accent color: `[a]_part_xyz.stl`
    * Opaque color (Blocks light): `[o]_part_xyz.stl`
    * Clear/transparent color (Allows light): `[c]_part_xyz.stl`
    * Quantity, if more than one is needed: `part_xyz_x4.stl`
  * [ x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [ x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [ x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x ] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [ x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [ x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [ x] Submitted STL files are oriented and scaled properly for printing.
  * [ x] Submission includes a CAD file in the form of a `.STEP` or `.SCAD` file
  * [ x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?
Modified stealthburner dragon toolhead and associated files to add a piezo disc as Z probe.

#### How was it tested?
I'm using it on my Trident 300. Video is provided.

#### Any background context you want to provide?
I added precision piezo probing to my i3Bear under the bed and never want to go back to inductive. So when building my Trident, I wanted to have piezo as well. I have not found the piezo rings for ultrasonic applications to be used for probing, so the project was a little risky. I was very happy the solution works so well.

#### Screenshots (if appropriate)
N/A

#### Further notes
My hardware was made with SB beta, but I re-edited the toolhead files with SB release.

### CODE REVIEW

**Comment by MapleLeafMakers** [comment]
CAD Files are missing for `stealthburner_printhead_dragon_rear_piezo.stl` and `stealthburner_printhead_dragon_front_piezo.stl`

**Comment by MapleLeafMakers** [comment]
There are STL errors in `stealthburner_printhead_dragon_rear_piezo.stl`

image

They are likely caused by this unusual geometry in the step file:

image

**Comment by 4k0b** [comment]
Fixed STL

**Comment by MapleLeafMakers** [comment]
The CAD Files also need some work.  The step file should able to be exported back to an STL and produce the same part, but all I can get out of it is this:
image

The step files are required so that people can make their own modifications, and since they aren't representative of the stl files, that's not really possible.

They also appear to have been imported originally from STL files rather than the Stealthburner CAD, which makes future modifications much harder, since all curved surfaces have been replaced with mesh approximations.  I'm not comfortable approving this as-is, and would recommend reworking the CAD, but I will let one of the maintainers make the final decision.

**Comment by 4k0b** [comment]
Redid CAD files and exchanged rear STL file. The CAD files are much smaller now and are based on CAD instead of mesh.

File: printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md
Comment (cbc02009) [inline printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md:5]: please use the relative link to the file here (./Images/Probing.mp4)

File: printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md
Comment (cbc02009) [inline printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md:31]: A link to a specific example on ali might be helpful here.

**Review by cbc02009 — CHANGES_REQUESTED** [review]
Please link a commit or list a version number in the readme for which version of stealthburner you used in case there are any breaking changes in new versions.

**Comment by 4k0b** [comment]
I don't see how to directly link the local mp4, it does not work. I followed https://github.blog/2021-05-13-video-uploads-available-github/

Currently on ali, there's no pzt8 with correct dimensions, but there are close matches. I added direct link to manufacturer and to ebay.

File: printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md
Comment (4k0b) [reply printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md]: tried, it's not working

Comment (4k0b) [reply printer_mods/4k0b/StealthburnerDragonPiezoProbing/README.md]: If you know how to insert the correct link, please feel free to do so.

**Pull request closed without merging**
